tests/rptest/tests/tx_verifier_test.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from ducktape.mark.resource import cluster
+from ducktape.errors import DucktapeError
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.clients.rpk import RpkTool
+
+import subprocess
+
+
+class TxVerifierTest(RedpandaTest):
+    """
+    Verify that segment indices are recovered on startup.
+    """
+    def __init__(self, test_context):
+        extra_rp_conf = {
+            "enable_idempotence": True,
+            "enable_transactions": True,
+            "default_topic_replications": 3,
+            "default_topic_partitions": 1
+        }
+
+        super(TxVerifierTest, self).__init__(test_context=test_context,
+                                             extra_rp_conf=extra_rp_conf)
+
+    @cluster(num_nodes=3)
+    def test_tx(self):
+        verifier_jar = "/opt/tx-verifier/tx-verifier.jar"
+
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic("topic1")
+        rpk.create_topic("topic2")
+
+        self.redpanda.logger.error("starting tx verifier")
+        try:
+            cmd = ("{java} -jar {verifier_jar} {brokers}").format(
+                java="java",
+                verifier_jar=verifier_jar,
+                brokers=self.redpanda.brokers())
```

Comment:
@rystsov @ivotron when writing ducktape tests we want to avoid as much as possible doing "work" on the orchestrator node. the solution to fix that in this test is to instantiate an instance of the ducktape background thread service and run the verifier in that service. that has the affect of allocating a node in the cluster specifically to run the verifier.

there are exceptions to the rule. the general rule of thumb is that i'ts ok to do basic stuff like admin operations, a little producing/consuming, etc... does verifier (or if we scaled up verifier to be bigger or longer) fit into this category?

if it should be a service can we make a ticket to address that? otherwise, we may be forced to address it on-demand when we begin scaling up ducktape integration testing.

Reply:
Created a ticket: https://github.com/vectorizedio/redpanda/issues/1918

Reply:
out-of-band: this particular verifier is intended to be quite small (generate not much load / few events) and is therefore reasonable to run inline to the orchestrator.